apps/web/test/lib/next-config.test.ts

```diff
@@ -1,173 +1,180 @@
 
-import { it, expect, describe, beforeAll, afterAll } from "vitest";
+import { it, expect, describe, beforeAll } from "vitest";
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const { getSubdomainRegExp } = require("../../getSubdomainRegExp");
-let userTypeRouteRegExp: RegExp;
-let teamTypeRouteRegExp:RegExp;
-let privateLinkRouteRegExp:RegExp
-let embedUserTypeRouteRegExp:RegExp
-let embedTeamTypeRouteRegExp:RegExp
-
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const {match, pathToRegexp} = require("next/dist/compiled/path-to-regexp");
+type MatcherRes = (path: string) => {params: Record<string, string>}
+let userTypeRouteMatch: MatcherRes;
+let teamTypeRouteMatch:MatcherRes;
+let privateLinkRouteMatch:MatcherRes
+let embedUserTypeRouteMatch:MatcherRes
+let embedTeamTypeRouteMatch:MatcherRes
+let orgUserTypeRouteMatch:MatcherRes
+let orgUserRouteMatch: MatcherRes
+
+
+beforeAll(async()=>{
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  process.env.NEXT_PUBLIC_WEBAPP_URL = "http://example.com"
+  const {
+    userTypeRoutePath,
+    teamTypeRoutePath,
+    privateLinkRoutePath,
+    embedUserTypeRoutePath,
+    embedTeamTypeRoutePath,
+    orgUserRoutePath,
+    orgUserTypeRoutePath,
+    orgUserTypeEmbedRoutePath,
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  } = require("../../pagesAndRewritePaths")
+
+  userTypeRouteMatch = match(userTypeRoutePath);
```

Comment:
No hardcoding of RegExps here now. We do an automatic conversion the way it's done by Next.js